src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponent.scala

```diff
@@ -0,0 +1,94 @@
+
+  import driver.api._
+
+  class SubmissionTable(tag: Tag) extends Table[SubmissionRecord](tag, "SUBMISSION") {
+    def id = column[UUID]("ID", O.PrimaryKey)
+    def submissionDate = column[Timestamp]("DATE_SUBMITTED")
+    def submitterId = column[String]("SUBMITTER")
+    def methodConfigurationId = column[Long]("METHOD_CONFIG_ID")
+    def submissionEntityId = column[Option[Long]]("ENTITY_ID")
+    def statusId = column[Long]("STATUS")
+
+    def * = (id, submissionDate, submitterId, methodConfigurationId, submissionEntityId, statusId) <> (SubmissionRecord.tupled, SubmissionRecord.unapply)
+
+    def submitter = foreignKey("FK_SUB_SUBMITTER", submitterId, rawlsUserQuery)(_.userSubjectId)
+    def methodConfiguration = foreignKey("FK_SUB_METHOD_CONFIG", methodConfigurationId, methodConfigurationQuery)(_.id)
+    def submissionEntity = foreignKey("FK_SUB_ENTITY", submissionEntityId, entityQuery)(_.id)
+    def status = foreignKey("FK_SUB_STATUS", statusId, submissionStatusQuery)(_.id)
+  }
+
+  class SubmissionStatusTable(tag: Tag) extends Table[SubmissionStatusRecord](tag, "SUBMISSION_STATUS") {
+    def id = column[Long]("ID", O.PrimaryKey, O.AutoInc)
+    def name = column[String]("NAME")
+
+    def * = (id, name) <> (SubmissionStatusRecord.tupled, SubmissionStatusRecord.unapply)
+
+    def statusNameIdx = index("IDX_SUB_STATUS_NAME", (name), unique = true)
+  }
+
+  class SubmissionWorkflowTable(tag: Tag) extends Table[SubmissionWorkflowRecord](tag, "SUBMISSION_WORKFLOW") {
+    def submissionId = column[UUID]("SUBMISSION_ID")
+    def workflowId = column[UUID]("WORKFLOW_ID")
+
+    def * = (submissionId, workflowId) <> (SubmissionWorkflowRecord.tupled, SubmissionWorkflowRecord.unapply)
+
+    def submission = foreignKey("FK_SUB_WF_SUB", submissionId, submissionQuery)(_.id)
+    def workflow = foreignKey("FK_SUB_WF_WF", workflowId, workflowQuery)(_.id)
+  }
+
+  class SubmissionWorkflowFailureTable(tag: Tag) extends Table[SubmissionWorkflowFailureRecord](tag, "SUBMISSION_WORKFLOW_FAILURE") {
+    def submissionId = column[UUID]("SUBMISSION_ID")
+    def workflowFailureId = column[Long]("WORKFLOW_FAILURE_ID")
+
+    def * = (submissionId, workflowFailureId) <> (SubmissionWorkflowFailureRecord.tupled, SubmissionWorkflowFailureRecord.unapply)
+
+    def submission = foreignKey("FK_SUB_WF_FAILURE_SUB", submissionId, submissionQuery)(_.id)
+    def workflowFailure = foreignKey("FK_SUB_WF_FAILURE_WF", workflowFailureId, workflowFailureQuery)(_.id)
+  }
+
+  class SubmissionValidationTable(tag: Tag) extends Table[SubmissionValidationRecord](tag, "SUBMISSION_VALIDATION") {
+    def workflowId = column[UUID]("WORKFLOW_ID")
+    def valueId = column[Option[Long]]("VALUE_ID")
+    def errorText = column[Option[String]]("ERROR_TEXT")
+    def inputName = column[String]("INPUT_NAME")
+
+    def * = (workflowId, valueId, errorText, inputName) <> (SubmissionValidationRecord.tupled, SubmissionValidationRecord.unapply)
```

Comment:
Per the domain object transition doc:

SUBMISSION_VALIDATION_VALUE // to case class
workflow: FK to WORKFLOW.id / WORKFLOW_FAILURE.id (does this work?)
value: FK to ATTRIBUTE.id (nullable)
error: String (nullable)
inputName: String

What should the approach be here? I've included just workflowId for the sake of having something that compiles. Should it be something like: workflowId: Option[UUID], workflowFailureId: Option[Long] ?